src/app/games/checkers/common/tests/CheckersMove.spec.ts

```diff
@@ -0,0 +1,212 @@
+
+            // When calling isPrefix on one and passing the other
+            // Then the result should be false
+            expect(long.isPrefix(short)).toBeTrue();
+            expect(short.isPrefix(long)).toBeTrue();
+        });
+
+        it('should not consider equal move as prefix to each others', () => {
+            // Given two different moves
+            const first: CheckersMove = CheckersMove.fromCapture([new Coord(2, 2), new Coord(4, 4)]).get();
+            const second: CheckersMove = CheckersMove.fromCapture([new Coord(2, 2), new Coord(4, 4)]).get();
+
+            // When calling isPrefix on them
+            // Then the result should be false
+            expect(first.isPrefix(second)).toBeFalse();
+        });
+
+    });
+
+    describe('getStartingCoord', () => {
+
+        it('should return the first coord', () => {
+            // Given any move
+            const move: CheckersMove = CheckersMove.fromCapture([new Coord(2, 2), new Coord(4, 4)]).get();
+
+            // When calling getStartingCoord
+            const startingCoord: Coord = move.getStartingCoord();
+
+            // Then it should return first coord
+            expect(startingCoord.equals(move.coords.get(0))).toBeTrue();
+        });
+
+    });
+
+    describe('getEndingCoord', () => {
+
+        it('should return the last coord', () => {
+            // Given any move
+            const move: CheckersMove = CheckersMove.fromCapture([new Coord(2, 2), new Coord(4, 4)]).get();
+
+            // When calling getStartingCoord
+            const endingCoord: Coord = move.getEndingCoord();
+
+            // Then it should return first coord
+            expect(endingCoord.equals(move.coords.get(1))).toBeTrue();
+        });
+
+    });
+
+    describe('getSteppedOverCoords', () => {
+
+        it('should return the coords between move.coords', () => {
+            // Given a capture
+            const move: CheckersMove = CheckersMove.fromCapture([new Coord(2, 2), new Coord(4, 4)]).get();
+
+            // When calling getSteppedOverCoords
+            const steppedOverCoords: Coord[] = move.getSteppedOverCoords().get().toList();
+
+            // Then the piece should be the stepped over coords
+            expect(steppedOverCoords).toEqual([new Coord(2, 2), new Coord(3, 3), new Coord(4, 4)]);
+        });
+
+    });
+
+    describe('concatenate', () => {
+
+        it('should concatenate moves and return a new one', () => {
+            // Given two moves, the second starting where the first start
+            const first: CheckersMove = CheckersMove.fromCapture([new Coord(0, 0), new Coord(2, 2)]).get();
+            const second: CheckersMove = CheckersMove.fromCapture([new Coord(2, 2), new Coord(4, 4)]).get();
+
+            // When concatenating them
+            const third: CheckersMove = first.concatenate(second);
+
+            // Then the third should have all the coords
```

Comment:
"given two moves, then the third", eeeeh
--> "then their concatenation should ..."
ou "when concatenating them into a third move"